hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3,18 +3,1146 @@
     {
-      name: sanity
-      desc: ''' Basic hello world type test.
+      name: chip_uart_tx_rx
+      desc: '''Verify transmission of data over the TX and RX port.
+
+            SW test sends a known payload over the TX port. The testbench, at the same time
+            sends a known payload over RX. On reception, both payloads are checked for integrity.
+            SW validates the reception of TX watermark, RX watermark, and the TX empty interrupts.
+            Verify each UART instance at the chip level independently.
             '''
       milestone: V1
-      tests: ["chip_sanity"]
+      tests: ["chip_uart_tx_rx"]
+    }
+    {
+      name: chip_uart_rx_overflow
+      desc: '''Verify the RX overflow interrupt.
+
+            The testbench sends a random payload of size greater than the RX fifo size (32). The SW
+            ignores the received the data to allow the RX overflow interrupt to assert.
+            Verify each UART instance at the chip level independently.
+            '''
+      milestone: V1
+      tests: ["chip_uart_tx_rx"]
+    }
+
+    // GPIO (pre-verified IP) integration tests:
+    {
+      name: chip_gpio_out
+      desc: '''Verify GPIO outputs.
+
+            SW test configures the GPIO to be in the output mode. The test walks a 1 through the
+            pins. The testbench checks the value for correctness.
+            '''
+      milestone: V1
+      tests: []
+    }
+    {
+      name: chip_gpio_in
+      desc: '''Verify GPIO inputs.
+
+            The SW test configures the GPIO to be in input mode and enables all of them to generate
+            an interrupt. The testbench walks a 1 through the pins. SW test ensures that the
+            interrupt corresponding to the right pin is seen.
+            '''
+      milestone: V1
+      tests: []
+    }
+
+    // SPI_DEVICE (pre-verified IP) integration tests:
+    {
+      name: chip_spi_device_tx_rx
+      desc: '''Verify the transmission of data on the chip's SPI device port.
+
+            The testbench sends a known payload over the chip's SPI device input port. The SW test,
+            at the same time sends a known payload out over the chip's SPI device output port. On
+            reception, both payloads are checked for integrity. SW validates the reception of RX
+            fifo full, RX fifo over level, TX fifo under level, RX overflow and TX underflow
+            interrupts.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_spi_device_csb
+      desc: '''Verify the SPI device chip select pin.
+
+            The testbench attempts to send payload while the chip select pin is 1 (chip not
+            selected). SW ensures that no transmission occurred.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_spi_device_eeprom
+      desc: '''Verify the SPI device in EEPROM mode.
+
+            SW puts the SPI device in EEPROM mode and allows a firmware image (a.k.a. a known
```

Comment:
is this supposed to be like the boot-strap mode in ROM?

Reply:
Not sure of the right answer to this question. I am thinking yes? 
@gkelly